src/migrate-config-options.js

```diff
@@ -0,0 +1,104 @@
+'use babel'
+
+/*
+ * These migrations can take one of two forms, a direct move or a general function.
+ *
+ * Direct move:
+ *   These objects have a `type` of `direct-move`, and an array of `moves`, which
+ *   are objects containing an `old` setting name and a `new` setting name.
+ *   Any existing config found in the `old` name will be moved over to (and overwrite)
+ *   the `new` key.
+ *
+ * Functions:
+ *   These have a `type` of `function`, and a `migrationFunc`, which takes the
+ *   current linter-eslint atom config as an argument, and can act on it however
+ *   it needs to.
+ */
+const migrations = [
+  {
+    type: 'direct-move',
```

Comment:
I think the direct move is more simple and clear for this, but it is a bit annoying to need a function type also because of the not-direct migration from earlier. Since we're running a function anyway, I guess I agree it seems better to go ahead and handle it inside the function.

Definitely agree with the rename.